cli/utilities/containers.py

```diff
@@ -53,72 +53,78 @@ def __init__(self, nodes, package="podman"):
 
     def run(
         self,
+        name=None,
         image=None,
         rm=None,
         env=None,
         volume=None,
         ports=None,
-        detach_key=None,
         restart=None,
-        detach=False,
+        detach=None,
         cmds=None,
     ):
         """Executes the provided command using podman
         Args
+            name (str): Container name
             image (str): Image name
             env (list): List of environment variables
             volume (list): List of volumes
             ports (list): List of ports
             rm (list): List of entries to be removed
-            detach (bool): Detach flag
+            detach (str): Detach flag
             restart (str): Restart flag
             cmds (str): Other commands to be executed
         """
-        if not image and not rm:
+        if not name and not rm:
```

Comment:
Why this change ? which image is mandatory, not name

Reply:
Yes, you are correct, based on the general run command, IMAGE is a mandatory parameter:
``` podman run [options] IMAGE [COMMAND [ARG...]] ```

I've some queries regarding options we are currently using for podman run command for disconnected installation:

We are passing image with different parameters for e.g.  --detach IMAGE (Basically to run the container in background) and --rm IMAGE (To remove container when stopped)

 How should we handle such scenarios with the current implementation?

Reply:
@adityaramteke as per documentation [1]. 

`name` is not a mandatory parameter. `image` is mandatory and `rm` is optional parameter which requires `image` to be passed.
Also in case  `detach` parameter doesn't have any value then it works as `-d` (which will run container in background) but here `detach` is passed as list or string as `--detach` it accepts multiple string, here we `IMAGE` is used because it's passed at the end.

[1] https://docs.podman.io/en/latest/markdown/podman-run.1.html

Reply:
Thanks, Vaibhav! I've made the changes.